libraries/AC_PID/AC_PID_2D.cpp

```diff
@@ -215,33 +201,65 @@ void AC_PID_2D::save_gains()
     _kp.save();
     _ki.save();
     _kd.save();
-    _imax.save();
-    _filt_hz.save();
-    _filt_d_hz.save();
+    _kff.save();
+    _kimax.save();
+    _filt_E_hz.save();
+    _filt_D_hz.save();
 }
 
-// calc_filt_alpha - recalculate the input filter alpha
-void AC_PID_2D::calc_filt_alpha()
+/// Overload the function call operator to permit easy initialisation
+void AC_PID_2D::operator()(float p_val, float i_val, float d_val, float ff_val, float imax_val, float input_filt_E_hz, float input_filt_D_hz, float dt)
 {
-    if (is_zero(_filt_hz)) {
-        _filt_alpha = 1.0f;
-        return;
-    }
-  
-    // calculate alpha
-    const float rc = 1/(M_2PI*_filt_hz);
-    _filt_alpha = _dt / (_dt + rc);
+    _kp = p_val;
+    _ki = i_val;
+    _kd = d_val;
+    _kff = ff_val;
+    _kimax = fabsf(imax_val);
+    _filt_E_hz = input_filt_E_hz;
+    _filt_D_hz = input_filt_D_hz;
+    _dt = dt;
+}
+
+// get_filt_T_alpha - get the target filter alpha
+float AC_PID_2D::get_filt_E_alpha() const
+{
+    return get_filt_alpha(_filt_E_hz);
+}
+
+// get_filt_D_alpha - get the derivative filter alpha
+float AC_PID_2D::get_filt_D_alpha() const
+{
+    return get_filt_alpha(_filt_D_hz);
 }
 
-// calc_filt_alpha - recalculate the input filter alpha
-void AC_PID_2D::calc_filt_alpha_d()
+// get_filt_alpha - calculate a filter alpha
+float AC_PID_2D::get_filt_alpha(float filt_hz) const
```

Comment:
I think we're supposed to be using `calc_lowpass_alpha_dt` now-adays.

It does at least have another sanity check that _dt is > 0.

Looks like this was already done in the 1D variant.

Reply:
txs, changed to use calc_lowpass_alpha_dt